saskatoon/harvest/api.py

```diff
@@ -220,3 +223,170 @@ def list(self, request, *args, **kwargs):
                                  'title': _("New Person")
                                  }
                          })
+
+
+class StatsView(LoginRequiredMixin, generics.ListAPIView):
+    template_name = "app/stats.html"
+    queryset = Harvest.objects.filter(status="Succeeded")
+    filter_backends = (filters.DjangoFilterBackend,)
+    filterset_class = HarvestFilter
+    filterset_fields = ('status', 'season')
+    permission_classes = [IsCoreOrAdmin]
+
+    def list(self, request, format="html", *args, **kwargs):
+        """returns statistics on harvests for all seasons or a specific season"""
+        season = self.request.GET['season']
+        error_message = f"No harvests were found for {season=}. Displaying all statistics..."
+
+        self.harvest_queryset = self.filter_queryset(self.get_queryset())
+        #? Show error message and display all stats if the queryset is empty
+        if not self.harvest_queryset:
+            self.harvest_queryset = self.get_queryset()
+            messages.error(self.request, error_message, "danger")
+            #? display 'All' on the dropdown button instead of the specified season to avoid confusion
+            season = 'All'
+
+        self.harvest_yield_queryset = HarvestYield.objects.filter(harvest__in=self.harvest_queryset)
```

Comment:
First I think it'd maybe make more sense to define error_message under the if statement:
 ```
        self.harvest_queryset = self.filter_queryset(self.get_queryset())
        if not self.harvest_queryset:
            self.harvest_queryset = self.get_queryset()
            error_message = f"No harvests were found for {season=}. Displaying all statistics..."
            messages.error(self.request, error_message, "danger")
            season = 'All'
 ```

But then again your queryset could also be empty because of the `status` field available in your filter, then your message wouldn't really be accurate (same for other filters we might want to add in the future).
Also what if for some reason `self.get_queryset()` is empty (e.g. fresh database) ? It sure is an edge case but getting a server error could be avoided.
      
Are the `aggregate(Sum("field_name"))` bits throwing errors on empty querysets? 
      
I personally think it'd make more sense to present an empty page, that is that each of your function should handle  an empty queryset and return 0.  
Or you could also simply return 0 for all of them right away, definitely not the most elegant though, e.g.:

```
      if not self.harvest_queryset:
      
          no_highlights = {"total_harvests": 0,
                               "total_pickers": 0,
                                "total_beneficiaries": 0,
                                "total_weight": 0 }
                                
              return Response(
            {
                "season": season,
                "seasons": self.filterset_class.YEARS,
                "highlights": no_hightlights,
                "total_fruit": 0,
                "total_neighborhood": 0,
                "total_beneficiary": 0,
                "total_picker": 0,
            }
        )
```
     
If it's too much of a hassle we can keep your solution for now

Reply:
Thanks for pointing all this out, very valid points!
>Are the aggregate(Sum("field_name")) bits throwing errors on empty querysets?

Yes that's where I get an error:
```sh
Exception Type: TypeError at /stats/
Exception Value: int() argument must be a string, a bytes-like object or a number, not 'NoneType'
```
--------
For empty data that's what gets displayed in the tables:

![image](https://user-images.githubusercontent.com/52796958/168929793-ea5cfc8f-8a1e-4483-8e69-109a855deb57.png)

Is that good, or would adding a single row with `N/A` data would be more good looking?.

Reply:
yes that's totally fine! Why do you still get non null values in all other tables though?

Reply:
I've added this harvest through the admin panel. It probably let me add the harvest without adding what trees the picker have picked.

Reply:
![image](https://user-images.githubusercontent.com/52796958/168933144-547f9a19-831a-4c99-be0e-c351ba1bafc7.png)
It happens because I didn't add a person as a recipient.

![image](https://user-images.githubusercontent.com/52796958/168933306-597159ac-6d76-4d11-a37d-4c50db9bfdd6.png)